test/unit/test-url-replacements.js

```diff
@@ -595,6 +605,12 @@ describes.sandboxed('UrlReplacements', {}, () => {
     });
   });
 
+  it('should replace PAGE_VIEW_ID_64', () => {
+    return expandUrlAsync('?pid=PAGE_VIEW_ID_64').then(res => {
+      expect(res).to.match(/pid=([a-zA-Z0-9_-]+){10,}/);
```

Comment:
can we add a check to the res length is 64. Thanks

Reply:
@zhouyx Why should it be a string of 64 characters length ? It will try to base64 encode  a string of 16 characters (We use `window.crypto.getRandomValues(<Uint8array>)` until we generates randomly 128bits).  Each character is used to represent 6 bits (log2(64) = 6). Therefore 4 chars are used to represent 4 * 6 = 24 bits = 3 bytes. 16 is not a multiple of 3 so we add a padding of 2. Now we have 4 * (16 + 2)/3 = 24 if I'm not wrong. 
Did I got the wrong implementation ?

Reply:
I'm sorry you're right. I was thinking about the crypto.sha384Base64 fallback. 

This looks great. Thanks